File: app/src/main/java/sugtao4423/lod/userpage_fragment/UserPageFragmentPagerAdapter.java
package sugtao4423.lod.userpage_fragment;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import sugtao4423.lod.R;

public class UserPageFragmentPagerAdapter extends FragmentStatePagerAdapter{

    private Context context;

    private _0_detail detail;
    private _1_Tweet tweet;
    private _2_favorites favorites;
    private _3_follow follow;
    private _4_follower follower;

    public UserPageFragmentPagerAdapter(FragmentManager fm, Context context){
        super(fm);
        this.context = context;

        detail = new _0_detail();
        tweet = new _1_Tweet();
        favorites = new _2_favorites();
        follow = new _3_follow();
        follower = new _4_follower();
    }

    @Override
    public Fragment getItem(int i){
        switch(i){
            case 0:
                return detail;
            case 1:
                return tweet;
            case 2:
                return favorites;
            case 3:
                return follow;
            case 4:
                return follower;
            default:
                return null;
        }
    }

    @Override
    public int getCount(){
        return 5;
    }

    @Override
    public CharSequence getPageTitle(int position){
        switch(position){
            case 0:
                return context.getString(R.string.page_title_detail);
            case 1:
                return context.getString(R.string.page_title_tweet);
            case 2:
                return context.getString(R.string.page_title_favorite);
            case 3:
                return context.getString(R.string.page_title_follow);
            case 4:
                return context.getString(R.string.page_title_follower);
            default:
                return null;
        }
    }

}

File: app/src/main/java/sugtao4423/lod/userpage_fragment/UserPageFragmentPagerAdapter.kt
package sugtao4423.lod.userpage_fragment

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import sugtao4423.lod.R
import twitter4j.User

class UserPageFragmentPagerAdapter(fm: FragmentManager, private val context: Context) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val detail = _0_detail()
    private val tweet = _1_Tweet()
    private val favorites = _2_favorites()
    private val follow = _3_follow()
    private val follower = _4_follower()

    override fun getItem(i: Int): Fragment {
        return when (i) {
            1 -> tweet
            2 -> favorites
            3 -> follow
            4 -> follower
            else -> detail
        }
    }

    override fun getCount(): Int {
        return 5
    }

    override fun getPageTitle(position: Int): CharSequence {
        return when (position) {
            1 -> context.getString(R.string.page_title_tweet)
            2 -> context.getString(R.string.page_title_favorite)
            3 -> context.getString(R.string.page_title_follow)
            4 -> context.getString(R.string.page_title_follower)
            else -> context.getString(R.string.page_title_detail)
        }
    }

    fun setTargetUser(user: User) {
        detail.targetUser = user
        detail.setText()
        tweet.targetUser = user
        favorites.targetUser = user
        follow.targetUser = user
        follower.targetUser = user
    }

}